File: authhub-ui/src/app/models/AuthSettings.ts
import { AuthSchemeEnum } from "./AuthSchemeEnum";

export class AuthSettings {
    Id: number | null | undefined; // Assuming EntityBase<int> translates to an `id` property
    Name: string | null | undefined;
    AuthScheme: AuthSchemeEnum | null | undefined; // Ensure AuthScheme is defined elsewhere
    SaltLength: number | null | undefined;
    HashLength: number | null | undefined;
    Iterations: number | null | undefined;
    Key: string | null | undefined; // You might want to handle MinLength validation in your forms
    Issuer: string | null | undefined;
    Audience: string | null | undefined;
    ExpirationMinutes: number | null | undefined; // Optional property with default value

    constructor(init?: Partial<AuthSettings>){
        Object.assign(this, init);
        //TODO: this is hardcoded and probably bad practice, but let's just do it for now
        this.AuthScheme = AuthSchemeEnum.JWT;
    }
}

File: AuthHub.Models/Entities/Organizations/AuthSettings.cs
using AuthHub.Models.Entities.Enums;
using Common.Models.Entities;
using System.ComponentModel.DataAnnotations;

namespace AuthHub.Models.Entities.Organizations
{
    public class AuthSettings : EntityBase<int>
    {
        public int OrganizationID { get; set; }
        public AuthScheme AuthScheme { get; set; }
        public int AuthSchemeID { get; set; }
        public int SaltLength { get; set; }
        public int HashLength { get; set; }
        public int Iterations { get; set; }
        [MinLength(8)]
        public string Key { get; set; }
        public string Issuer { get; set; }
        public string Audience { get; set; }
        public int ExpirationMinutes { get; set; } = 30;
        public string PasswordResetUrl { get; set; }
        public Organization Organization { get; set; }
    }
}